File: cpp/src/smallsword.cpp
/******************************************************************************
 * AUTHOR:      Alexander Casal
 * FILE:        SmallSword.cpp
 * DESCRIPTION: SmallSword is a small weapon which will always hit the enemy.
 *              The amount of damage is small.
 *****************************************************************************/
#include "smallsword.h"
#include <iostream>

/**
* Constructor
*/
SmallSword::SmallSword()
{
	std::cout << "Small sword equipped\n";
}

/**
* attack
*
* This is where we implement a specific attack strategy for our Weapon.
* In this case, it is a simple notification that the enemy received damage.
*/
void SmallSword::attack()
{
	std::cout << "You swing your sword. The enemy is hurt\n";
}

File: java/SmallSword.java
package strategygame;

/**
 * SmallSword is a small weapon which will always hit the enemy. The
 * amount of damage is small.
 * @author Alexander
 */
public class SmallSword implements Weapon {
    
    /**
     * Constructor
     * 
     * Notify we have equipped a Small Sword
     */
    public SmallSword() {
        
        System.out.println("Small sword equipped");
    }
    
    /**
     * attack
     * 
     * Swing the sword and hurt the enemy
     */
    public void attack() {
        
        System.out.println("You swing your sword. The enemy is hurt");
    }
}
